spring-cloud-gcp-logging/src/main/java/org/springframework/cloud/gcp/logging/LoggingWebMvcConfigurer.java

```diff
@@ -25,18 +27,22 @@
  * MVC Adapter that adds the {@link TraceIdLoggingWebMvcInterceptor}
  *
  * @author Mike Eltsufin
+ * @author Chengyuan Zhao
  */
 @Configuration
 public class LoggingWebMvcConfigurer extends WebMvcConfigurerAdapter {
 
 	private final TraceIdLoggingWebMvcInterceptor interceptor;
 
-	public LoggingWebMvcConfigurer(@Autowired(required = false) TraceIdLoggingWebMvcInterceptor interceptor) {
+	public LoggingWebMvcConfigurer(
+			@Autowired(required = false) TraceIdLoggingWebMvcInterceptor interceptor) {
 		if (interceptor != null) {
 			this.interceptor = interceptor;
 		}
 		else {
-			this.interceptor = new TraceIdLoggingWebMvcInterceptor();
+			this.interceptor = new TraceIdLoggingWebMvcInterceptor(
```

Comment:
How about having two different constructors?
One with and another one without the argument.

Reply:
Sounds good

Reply:
*ping*

Reply:
```
public LoggingWebMvcConfigurer() {
	this.interceptor = new TraceIdLoggingWebMvcInterceptor(
					new CompositeTraceIdExtractor(ImmutableList.of(
							new XCloudTraceIdExtractor(), new ZipkinTraceIdExtractor())));
}

public LoggingWebMvcConfigurer(TraceIdLoggingWebMvcInterceptor interceptor) {
	this.interceptor = interceptor;
}
```

Reply:
The `LoggingWebMvcConfigurer` is intended to be used from the `@Import` on the `@Configuration`, so those different ctors won't have any affects. More over I guess the default one will be selected by Spring Container for instantiation.